contracts/PetPark.sol

```diff
@@ -4,4 +4,93 @@ pragma solidity ^0.8.0;
 
 contract PetPark {
 
+    address  contractOwner;
+    enum AnimalType {
+        NONE, FISH, CAT, DOG, RABBIT, PARROT
+    }
+    enum Gender{
+        MALE, FEMALE
+    }
+    struct Animal {
+        AnimalType animalType;
+        uint count;
+    }
+
+    struct Borrower {
+        Gender gender;
+        uint age;
+        uint petCount;
+        bool exists;
+        AnimalType animalType;
+    }
+
+    event Added(AnimalType animalType, uint animalCount);
+    event Borrowed(AnimalType animalType);
+    event Returned(AnimalType animalType);
+
+    mapping(AnimalType => Animal) animals;
+    mapping(address => Borrower)borrowers;
+
+    constructor(){
+        contractOwner = msg.sender;
+    }
+
+    /*
+    * @dev method to add animals to the contract.@author Satish
+    */
+    function add(AnimalType _animalType, uint _count) external onlyOwner {
+        require(_animalType != AnimalType.NONE, "Invalid animal");
+        animals[_animalType].animalType = _animalType;
+        animals[_animalType].count += _count;
+        emit Added(_animalType, _count);
+    }
+
+    function borrow(uint _age, Gender _gender, AnimalType _animalType) external {
+        require(_age > 0, "Invalid Age");
+        require(_animalType != AnimalType.NONE, "Invalid animal type");
+        require(animals[_animalType].count > 0, "Selected animal not available");
+        if (borrowers[msg.sender].exists) {
+            require(borrowers[msg.sender].age == _age, "Invalid Age");
```

Comment:
Since `borrowers[msg.sender]` is being accesses multiple times better to store it in a memory variable and make all reads from it